overlord/snapstate/autorefresh_gating.go

```diff
@@ -551,6 +551,18 @@ func createGateAutoRefreshHooks(st *state.State, affectedSnaps map[string]*affec
 	return ts
 }
 
+func conditionalAutoRefreshAffectedSnaps(t *state.Task) ([]string, error) {
+	var snaps map[string]*refreshCandidate
+	if err := t.Get("snaps", &snaps); err != nil {
+		return nil, fmt.Errorf("internal error: cannot get snaps to update for %s task %s", t.Kind(), t.ID())
+	}
+	names := make([]string, 0, len(snaps))
+	for sn := range snaps {
+		names = append(names, sn)
+	}
```

Comment:
it's not urgent but we want to leave a TODO about dropping snaps once we know the outcome of hooks. We'll probably do that by setting something else on the task once done to filter things.

Reply:
Indeed, added.